sharding/indexHashedNodesCoordinator.go

```diff
@@ -591,6 +627,56 @@ func (ihgs *indexHashedNodesCoordinator) GetConsensusWhitelistedNodes(
 	return shardEligible, nil
 }
 
+// UpdatePeersListAndIndex will update the list and the index for all peers
+func (ihgs *indexHashedNodesCoordinator) UpdatePeersListAndIndex() error {
+	ihgs.mutNodesConfig.RLock()
+	defer ihgs.mutNodesConfig.RUnlock()
+
+	nodesConfig, ok := ihgs.nodesConfig[ihgs.currentEpoch]
+	if !ok {
+		return ErrEpochNodesConfigDoesNotExist
+	}
+
+	nodesConfig.mutNodesMaps.RLock()
+	defer nodesConfig.mutNodesMaps.RUnlock()
+
+	return ihgs.updatePeersListAndIndex(nodesConfig)
+}
+
+// updatePeersListAndIndex will update the list and the index for all peers
+// should be called with mutex locked
+func (ihgs *indexHashedNodesCoordinator) updatePeersListAndIndex(nodesConfig *epochNodesConfig) error {
+	err := ihgs.updatePeerAccountsForGivenMap(nodesConfig.eligibleMap, core.EligibleList)
+	if err != nil {
+		return err
+	}
+
+	err = ihgs.updatePeerAccountsForGivenMap(nodesConfig.waitingMap, core.WaitingList)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ihgs *indexHashedNodesCoordinator) updatePeerAccountsForGivenMap(
+	peers map[uint32][]Validator,
+	list core.PeerType,
+) error {
+	for _, accountsPerShard := range peers {
+		for index, account := range accountsPerShard {
+			err := ihgs.listIndexUpdater.UpdateListAndIndex(string(account.PubKey()), string(list), index)
```

Comment:
I think we should add also the shard (update the shard as well)

Reply:
done